pkg/adaptor/plugin.go

```diff
@@ -0,0 +1,255 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package adaptor
+
+import (
+	"fmt"
+	"github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	"github.com/che-incubator/che-workspace-operator/pkg/common"
+	"github.com/che-incubator/che-workspace-operator/pkg/config"
+	metadataBroker "github.com/eclipse/che-plugin-broker/brokers/metadata"
+	brokerModel "github.com/eclipse/che-plugin-broker/model"
+	"github.com/eclipse/che-plugin-broker/utils"
+	corev1 "k8s.io/api/core/v1"
+	"strconv"
+	"strings"
+)
+
+func AdaptPluginComponents(workspaceId, namespace string, devfileComponents []v1alpha1.ComponentSpec) ([]v1alpha1.ComponentDescription, *corev1.ConfigMap, error) {
+	var components []v1alpha1.ComponentDescription
+
+	broker := metadataBroker.NewBroker(true)
+
+	metas, _, err := getMetasForComponents(devfileComponents)
+	if err != nil {
+		return nil, nil, err
+	}
+	plugins, err := broker.ProcessPlugins(metas)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, plugin := range plugins {
+		component, err := adaptChePluginToComponent(workspaceId, plugin)
+		if err != nil {
+			return nil, nil, err
+		}
+		// TODO: Alias for plugins seems to be ignored in regular Che
```

Comment:
Not really. From che.openshift.io:
```json
{
  "namespace": "sleshche",
  "temporary": false,
  "id": "workspace6wt7edpromg4w9w0",
  "status": "RUNNING",
  "runtime": {
    "machines": {
      "nodejs": {
        "attributes": {
          "component": "nodejs",
          "memoryRequestBytes": "209715200",
          "memoryLimitBytes": "1073741824",
          "source": "recipe",
          "cpuLimitCores": "2.0",
          "cpuRequestCores": "0.125"
        }
      }
    },
    "commands": [
      {
        "commandLine": "yarn run lint",
        "name": "lint",
        "attributes": {
          "componentAlias": "nodejs",
          "machineName": "nodejs",
          "workingDir": "${CHE_PROJECTS_ROOT}/angular-realworld-example-app"
        },
        "type": "exec"
      }
    ]
  },
  "devfile": {
    "metadata": {
      "name": "wksp-sek5"
    },
    "components": [
      {
        "mountSources": true,
        "endpoints": [
          {
            "name": "angular",
            "port": 4200
          }
        ],
        "memoryLimit": "1Gi",
        "type": "dockerimage",
        "alias": "nodejs",
        "image": "quay.io/eclipse/che-nodejs10-community:nightly"
      }
    ],
    "apiVersion": "1.0.0",
    "commands": [
      {
        "name": "lint",
        "actions": [
          {
            "workdir": "${CHE_PROJECTS_ROOT}/angular-realworld-example-app",
            "type": "exec",
            "command": "yarn run lint",
            "component": "nodejs"
          }
        ]
      }
    ]
  }
}
```

So, alias is propagated to machine's and commands attributes. Then Theia use them to match command to container.
I'm OK with skipping alias for time being but it would be good to make comment up to date.

Reply:
Yes I'm fine to update this -- it was on my TODO list for Friday that I didn't get to. I recognize that alias is used for dockerimage components, but I could not find it being used for plugin components. I need to look into it more deeply.